### PR TITLE
Support a creation of pull request reviews

### PR DESCRIPTION
**Pull request by kusaeva**
See an [issue](https://github.com/47degrees/github4s/issues/563)

### CODE REVIEW

File: github4s/src/main/scala/github4s/domain/PullRequest.scala [hunk 1]
```diff
@@ -122,3 +122,29 @@ final case object PRRStateChangesRequested extends PullRequestReviewState("CHANG
...
+sealed abstract class PullRequestReviewEvent(val value: String)
+
+final case object PRREventApprove        extends PullRequestReviewEvent("APPROVE")
+final case object PRREventRequestChanges extends PullRequestReviewEvent("REQUEST_CHANGES")
+final case object PRREventComment        extends PullRequestReviewEvent("COMMENT")
+final case object PRREventPending        extends PullRequestReviewEvent("PENDING")
+
+case class CreateReviewComment(path: String, position: Int, body: String)
+
+case class ReviewComment(
```
Comment (BenFradet) [inline github4s/src/main/scala/github4s/domain/PullRequest.scala:142]: where is this case class used?

File: microsite/docs/pull_request.md [hunk 2]
```diff
@@ -219,7 +220,36 @@ The `result` on the right is the matching [PullRequestReview][pr-scala].
 
 See [the API doc](https://developer.github.com/v3/pulls/reviews/#get-a-single-review) for full reference.
 
+[pr-scala]: https://github.com/47degrees/github4s/blob/master/github4s/src/main/scala/github4s/domain/PullRequest.scala
```
Comment (BenFradet) [inline microsite/docs/pull_request.md:223]: nit: we usually leave link at the end of the document :+1:

File: microsite/docs/pull_request.md [hunk 3]
```diff
@@ -219,7 +220,36 @@ The `result` on the right is the matching [PullRequestReview][pr-scala].
...
+
+```scala mdoc:compile-only
+import github4s.domain.{CreatePRReviewRequest, PRREventApprove}
+
+val createReviewData = gh.pullRequests.createReview(
+  "47deg",
+  "github4s",
+  139,
+  CreatePRReviewRequest(Some("commit_id"), "body", PRREventApprove)  
+ )
```
Comment (BenFradet) [inline microsite/docs/pull_request.md:244]:
```suggestion
)
```

**Review by BenFradet — COMMENTED** [review]
Looks great, thanks you :+1: 
I just have a couple of nits.

File: github4s/src/main/scala/github4s/domain/PullRequest.scala (hunk 1 above)
Comment (kusaeva) [reply github4s/src/main/scala/github4s/domain/PullRequest.scala]: you're right, I removed it

**Comment by kusaeva** [comment]
What should I do to pass the code coverage test?

**Comment by BenFradet** [comment]
The integration tests can only be run by maintainers, that's why you're missing a few percentage points :+1:

File: microsite/docs/pull_request.md [hunk 4]
```diff
@@ -219,6 +220,35 @@ The `result` on the right is the matching [PullRequestReview][pr-scala].
...
+  139,
+  CreatePRReviewRequest(Some("commit_id"), "body", PRREventApprove)  
+)
+val response = createReviewData.unsafeRunSync()
+response.result match {
+  case Left(e) => println(s"Something went wrong: ${e.getMessage}")
+  case Right(r) => println(r)
+}
+```
+
```
Comment (BenFradet) [inline microsite/docs/pull_request.md:249]:
```suggestion

The `result` on the right is the created [PullRequestReview][pr-scala].

```

Also I think the `pr-scala` link got removed?

Comment (kusaeva) [reply microsite/docs/pull_request.md:249]: I moved it at the end of the document as you say. Or am I misreading you?

Comment (BenFradet) [reply microsite/docs/pull_request.md:249]: ah yes my bad it was already there.

**Review by BenFradet — APPROVED** [review]
Thanks :+1:

**Pull request merged**